### PR TITLE
#1: Fixing data model bugs related to JSON parsing exception by Gson.

### PR DESCRIPTION
**Pull request by alirezakhtm**

### CODE REVIEW

File: src/main/java/com/scayle/storefrontapi/model/Product.java [hunk 1]
```diff
@@ -91,30 +85,12 @@ public class Product implements ApiObjectInterface {
...
-    @SerializedName("pricePromotionInfo")
-    Object pricePromotionInfo;
-
     /**
     * 
     */
     @SerializedName("priceRange")
     PriceRange priceRange;
 
-    /**
```
Comment (HenriBeck) [inline src/main/java/com/scayle/storefrontapi/model/Product.java:106]: Why were these fields removed here?

Comment (alirezakhtm) [reply src/main/java/com/scayle/storefrontapi/model/Product.java:106]: These fields are not present in the JSON body that we receive from Scayle. When Gson tries to convert this JSON to a data model, exceptions occur.

Comment (HenriBeck) [reply src/main/java/com/scayle/storefrontapi/model/Product.java:106]: These fields are normally optional and will only be returned if they are requested via the `with` parameter.

Comment (HenriBeck) [reply src/main/java/com/scayle/storefrontapi/model/Product.java:106]: And you don't need these fields anywhere in your codebase?

Comment (alirezakhtm) [reply src/main/java/com/scayle/storefrontapi/model/Product.java:106]: I don't need it, but the problem lies elsewhere. When using this data model and converting JSON with Gson, it throws an exception. You can verify this. The field is not present in the JSON response from Scayle.

Comment (HenriBeck) [reply src/main/java/com/scayle/storefrontapi/model/Product.java:106]: It does have the field when you provide `with=reductionRange`

Comment (HenriBeck) [reply src/main/java/com/scayle/storefrontapi/model/Product.java:106]: So these fields are valid and this is why I'm hesitant to remove them here

Comment (alirezakhtm) [reply src/main/java/com/scayle/storefrontapi/model/Product.java:106]: Currently, we are using this dependency in our product and have practically discovered that this library needs to be corrected. I'm certain that these corrections are necessary. I strongly recommend checking this data model in connection with Scayle.

Comment (HenriBeck) [reply src/main/java/com/scayle/storefrontapi/model/Product.java:106]: https://dmg-live.storefront.api.scayle.cloud/v1/products?shopId=10001&with=reductionRange

This field does exist and as far as I know gson supports optional fields in the JSON response

**Pull request closed without merging**